File: contracts/js/utils/Block/block.js
module.exports.prevHash = function prevHash() {
  let prevhash = process.env.BLOCK_PREVHASH || "0";
  return BigInt(prevhash);
};

module.exports.number = function number() {
  let number = process.env.BLOCK_NUMBER || "0";
  return BigInt(number);
};

module.exports.chainId = function chainId() {
  let chainid = process.env.BLOCK_CHAINID || "0";
  return BigInt(chainid);
};

module.exports.coinbase = function coinbase() {
  let coinbase = process.env.BLOCK_COINBASE || "0";
  return BigInt(coinbase);
};

module.exports.difficulty = function difficulty() {
  let difficulty = process.env.BLOCK_DIFFICULTY || "0";
  return BigInt(difficulty);
};

module.exports.gasLimit = function gasLimit() {
  let gaslimit = process.env.BLOCK_GASLIMIT || "0";
  return BigInt(gaslimit);
};

module.exports.timestamp = function timestamp() {
  let timestamp = process.env.BLOCK_TIMESTAMP || "0";
  return BigInt(timestamp);
};

File: contracts/go/utils/Block/block.go
package block

import (
	"math/big"
	"os"

	common "qanx.space/qanx-contracts/go/utils/Common"
)

// Retrieves the prevHash of the block
func PrevHash() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_PREVHASH"))

	return n
}

// Retrieves the number of the block
func Number() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_NUMBER"))

	return n
}

// Retrieves the chain id of the block
func ChainId() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_CHAINID"))

	return n
}

// Retrieves the coinbase of the block
func Coinbase() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_COINBASE"))

	return n
}

// Retrieves the difficulty of the block
func Difficulty() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_DIFFICULTY"))

	return n
}

// Retrieves the gasLimit of the block
func GasLimit() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_GASLIMIT"))

	return n
}

// Retrieves the timestamp of the block
func Timestamp() *big.Int {
	n, _ := common.ParseBig256(os.Getenv("BLOCK_TIMESTAMP"))

	return n
}
